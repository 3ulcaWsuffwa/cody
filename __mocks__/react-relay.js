const ReactRelay = require("react-relay");

module.exports = {
  ...ReactRelay,
  // Mock createFragmentContainer HOC so it just returns the inner Component
  createFragmentContainer: Component => Component
};
