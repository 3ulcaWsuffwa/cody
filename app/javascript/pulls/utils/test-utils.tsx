import React from "react";
import { render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import { MemoryRouter } from "react-router";

// eslint-disable-next-line react/prop-types
const TestProvider = ({ children }) => {
  return (
    <IntlProvider locale="en" timeZone="UTC">
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </IntlProvider>
  )
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
function customRender(ui, options?) {
  return render(ui, { wrapper: TestProvider, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
