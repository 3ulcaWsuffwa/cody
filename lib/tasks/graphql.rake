namespace :graphql do
  desc "Export the GraphQL schema"
  task schema: :environment do
    File.write(
      Rails.root.join("schema.graphql"),
      GraphQL::Schema::Printer.print_schema(CodySchema)
    )
  end

  desc "Run Relay compiler"
  task relay: :schema do
    sh "bin/yarn relay"
  end
end
