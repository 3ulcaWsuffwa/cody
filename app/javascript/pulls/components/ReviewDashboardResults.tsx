import React from "react";
import {
  Calendar,
  ChevronRight,
  GitPullRequest,
  HelpCircle,
} from "react-feather";
import {
  createPaginationContainer,
  Environment,
  graphql,
  QueryRenderer,
  RelayPaginationProp,
} from "react-relay";
import Container from "./Container";
import List from "./List";
import { ReviewDashboardResults_viewer } from "./__generated__/ReviewDashboardResults_viewer.graphql";
import { ReviewDashboardResultsPullRequestExternalQuery } from "./__generated__/ReviewDashboardResultsPullRequestExternalQuery.graphql";
import { FormattedDate } from "react-intl";
import { Link } from "react-router-dom";
import LoadMore from "./LoadMore";

function ReviewDashboardResults({
  viewer,
  relay,
  environment
}: {
  viewer: ReviewDashboardResults_viewer;
  relay: RelayPaginationProp;
  environment: Environment
}): JSX.Element {
  return (
    <>
      <Container>
        <>
          <List>
            {() => {
              if (
                viewer.assignedReviews != null &&
                viewer.assignedReviews.edges != null &&
                viewer.assignedReviews.edges.length > 0
              ) {
                return (
                  <>
                    {viewer.assignedReviews.edges.map((edge) => {
                      if (edge != null && edge.node != null) {
                        const { node } = edge;
                        return (
                          <li key={node.id}>
                            <div className="flex items-center px-4 py-4 sm:px-6">
                              <div className="min-w-0 flex-1 flex items-center">
                                <div className="flex-shrink-0">
                                  <GitPullRequest />
                                </div>
                                <div className="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4">
                                  <div>
                                    <p className="text-sm font-medium truncate">
                                      <QueryRenderer<ReviewDashboardResultsPullRequestExternalQuery>
                                        environment={environment}
                                        query={graphql`
                                          query ReviewDashboardResultsPullRequestExternalQuery($nodeID: ID!) {
                                            node(id: $nodeID) {
                                              ...on PullRequest {
                                                title
                                                htmlUrl
                                              }
                                            }
                                          }
                                        `}
                                        variables={{ nodeID: node.pullRequest.id }}
                                        render={({ props, error }) => {
                                          if (error) {
                                            console.log(error.message);
                                            return (
                                              <span>{`${node.pullRequest.repository} #${node.pullRequest.number}`}</span>
                                            );
                                          } else if (props) {
                                            return (
                                              <a
                                                title={props.node.title}
                                                href={props.node.htmlUrl}
                                                rel="external"
                                                target="_blank"
                                              >
                                                {props.node.title}
                                              </a>
                                            );
                                          }
                                          return (
                                            <span>{`${node.pullRequest.repository} #${node.pullRequest.number}`}</span>
                                          );
                                        }}
                                      />
                                    </p>
                                    <div className="mt-2 sm:flex sm:justify-between">
                                      <div className="sm:flex">
                                        <p className="flex items-center text-sm text-gray-500">
                                          <span
                                            className="text-sm"
                                            title="Why was I assigned?"
                                          >
                                            <HelpCircle className="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400" />
                                          </span>
                                          {node.reviewRule != null
                                            ? node.reviewRule.name
                                            : "Manually assigned"}
                                        </p>
                                        <p className="mt-2 flex items-center text-sm text-gray-500 sm:mt-0 sm:ml-6">
                                          <span
                                            className="text-sm"
                                            title="Created at"
                                          >
                                            <Calendar className="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400" />
                                          </span>
                                          <FormattedDate
                                            value={
                                              new Date(
                                                node.pullRequest.createdAt
                                              )
                                            }
                                            year="numeric"
                                            month="short"
                                            day="2-digit"
                                          />
                                        </p>
                                      </div>
                                    </div>
                                  </div>
                                  <div className="hidden md:block">
                                    <div className="-space-x-1">
                                      {node.pullRequest.reviewers.nodes.map(
                                        (reviewer) => {
                                          return (
                                            <span
                                              key={reviewer.id}
                                              title={reviewer.login}
                                              className="relative inline-flex items-center justify-center h-6 w-6 rounded-full bg-gray-500 ring-2 ring-white"
                                            >
                                              <span className="text-xs font-medium leading-none text-white cursor-default">
                                                {reviewer.login
                                                  .charAt(0)
                                                  .toUpperCase()}
                                              </span>
                                              {reviewer.status ==
                                              "PENDING_APPROVAL" ? (
                                                <span className="absolute bottom-0 right-0 block h-1.5 w-1.5 rounded-full ring-2 ring-white bg-yellow-400"></span>
                                              ) : null}
                                            </span>
                                          );
                                        }
                                      )}
                                    </div>
                                    <div className="mt-2">
                                      <p className="text-sm text-gray-500">
                                        {`${node.pullRequest.repository} #${node.pullRequest.number}`}
                                      </p>
                                    </div>
                                  </div>
                                </div>
                                <div>
                                  <Link
                                    to={`/repos/${node.pullRequest.repository}/pull/${node.pullRequest.number}`}
                                    title="Open on Cody"
                                  >
                                    <ChevronRight />
                                  </Link>
                                </div>
                              </div>
                            </div>
                          </li>
                        );
                      }
                    })}
                  </>
                );
              } else {
                return (
                  <li>
                    <div className="mx-auto text-sm font-medium px-4 py-4 sm:px-6">
                      Nothing to show
                    </div>
                  </li>
                );
              }
            }}
          </List>
          <LoadMore relay={relay} />
        </>
      </Container>
    </>
  );
}

export default createPaginationContainer(
  ReviewDashboardResults,
  {
    viewer: graphql`
      fragment ReviewDashboardResults_viewer on User
      @argumentDefinitions(
        count: { type: "Int", defaultValue: 10 }
        cursor: { type: "String" }
        status: { type: ReviewerStatus, defaultValue: PENDING_APPROVAL }
        reviewRuleName: { type: "String" }
        repositoryName: { type: "String" }
      ) {
        assignedReviews(
          first: $count
          after: $cursor
          status: $status
          reviewRule: $reviewRuleName
          repository: $repositoryName
        ) @connection(key: "ReviewDashboardResults_assignedReviews") {
          edges {
            node {
              id
              reviewRule {
                name
              }
              pullRequest {
                id
                createdAt
                repository
                number
                reviewers {
                  nodes {
                    id
                    login
                    status
                  }
                }
              }
            }
          }
        }
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.viewer && props.viewer.assignedReviews;
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount
      }
    },
    getVariables(props, { count, cursor }, fragmentVariables) {
      return {
        count,
        cursor,
        reviewRuleName: fragmentVariables.reviewRuleName,
        repositoryName: fragmentVariables.repositoryName,
      };
    },
    query: graphql`
      query ReviewDashboardResultsPaginationQuery(
        $count: Int!
        $cursor: String
        $reviewRuleName: String
        $repositoryName: String
      ) {
        viewer {
          ...ReviewDashboardResults_viewer
            @arguments(
              count: $count
              cursor: $cursor
              reviewRuleName: $reviewRuleName
              repositoryName: $repositoryName
            )
        }
      }
    `,
  }
);