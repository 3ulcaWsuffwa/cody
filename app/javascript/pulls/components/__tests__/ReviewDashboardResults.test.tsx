import React from "react";
import { render, screen } from "test-utils";
import userEvent from "@testing-library/user-event";
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils";
import { Environment, graphql, QueryRenderer } from "react-relay";
import ReviewDashboardResults from "../ReviewDashboardResults";
import { ReviewDashboardResultsSnapshotQuery } from "./__generated__/ReviewDashboardResultsSnapshotQuery.graphql";

const TestComponent = ({ environment }: { environment: Environment }) => (
  <QueryRenderer<ReviewDashboardResultsSnapshotQuery>
    environment={environment}
    query={graphql`
      query ReviewDashboardResultsSnapshotQuery @relay_test_operation {
        viewer {
          ...ReviewDashboardResults_viewer
        }
      }
    `}
    variables={{}}
    render={({ error, props }) => {
      if (error) {
        return error.message;
      } else if (props) {
        return <ReviewDashboardResults environment={environment} viewer={props.viewer} />;
      } else {
        return "Loading";
      }
    }}
  />
);

let environment;
beforeEach(() => {
  environment = createMockEnvironment();
});

test("ReviewDashboardResults snapshot test", () => {
  const rendered = render(<TestComponent environment={environment} />);

  environment.mock.resolveMostRecentOperation((operation) =>
    MockPayloadGenerator.generate(operation, {
      ISO8601DateTime() {
        return "2020-11-30T00:33:49.973Z";
      },
      PageInfo() {
        return {
          hasNextPage: true,
          hasPreviousPage: false
        }
      }
    })
  );

  const fragment = rendered.asFragment();
  expect(fragment).toMatchSnapshot();

  environment.mock.resolveMostRecentOperation((operation) =>
    MockPayloadGenerator.generate(operation, {
      PullRequest() {
        return {
          title: "My Remote Title",
          htmlUrl: "https://example.local",
        };
      }
    })
  ); 

  const fragment_after_lazy_external_query = rendered.asFragment();
  expect(fragment_after_lazy_external_query).toMatchSnapshot();

  userEvent.click(screen.getByText("Load More"));
  const fragment_while_loading = rendered.asFragment();
  expect(fragment_while_loading).toMatchSnapshot();
});

test("ReviewDashboardResults snapshot test with no results", () => {
  const rendered = render(<TestComponent environment={environment} />);

  environment.mock.resolveMostRecentOperation((operation) =>
    MockPayloadGenerator.generate(operation, {
      ReviewerConnection() {
        return {
          edges: null,
        };
      },
    })
  );

  const fragment = rendered.asFragment();
  expect(fragment).toMatchSnapshot();
});
