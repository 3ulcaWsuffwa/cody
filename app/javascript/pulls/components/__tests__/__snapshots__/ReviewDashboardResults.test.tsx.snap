// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewDashboardResults snapshot test 1`] = `
<DocumentFragment>
  <div
    class="max-w-7xl mx-auto sm:px-6 lg:px-8 py-6"
  >
    <div
      class="bg-white shadow overflow-hidden sm:rounded-md"
    >
      <ul
        class="divide-y divide-gray-200"
      >
        <li>
          <div
            class="flex items-center px-4 py-4 sm:px-6"
          >
            <div
              class="min-w-0 flex-1 flex items-center"
            >
              <div
                class="flex-shrink-0"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="18"
                    cy="18"
                    r="3"
                  />
                  <circle
                    cx="6"
                    cy="6"
                    r="3"
                  />
                  <path
                    d="M13 6h3a2 2 0 0 1 2 2v7"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="9"
                    y2="21"
                  />
                </svg>
              </div>
              <div
                class="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4"
              >
                <div>
                  <p
                    class="text-sm font-medium truncate"
                  >
                    <span>
                      &lt;mock-value-for-field-"repository"&gt; #&lt;mock-value-for-field-"number"&gt;
                    </span>
                  </p>
                  <div
                    class="mt-2 sm:flex sm:justify-between"
                  >
                    <div
                      class="sm:flex"
                    >
                      <p
                        class="flex items-center text-sm text-gray-500"
                      >
                        <span
                          class="text-sm"
                          title="Why was I assigned?"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <circle
                              cx="12"
                              cy="12"
                              r="10"
                            />
                            <path
                              d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                            />
                            <line
                              x1="12"
                              x2="12.01"
                              y1="17"
                              y2="17"
                            />
                          </svg>
                        </span>
                        &lt;mock-value-for-field-"name"&gt;
                      </p>
                      <p
                        class="mt-2 flex items-center text-sm text-gray-500 sm:mt-0 sm:ml-6"
                      >
                        <span
                          class="text-sm"
                          title="Created at"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="18"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="4"
                            />
                            <line
                              x1="16"
                              x2="16"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="8"
                              x2="8"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="3"
                              x2="21"
                              y1="10"
                              y2="10"
                            />
                          </svg>
                        </span>
                        Nov 30, 2020
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden md:block"
                >
                  <div
                    class="-space-x-1"
                  >
                    <span
                      class="relative inline-flex items-center justify-center h-6 w-6 rounded-full bg-gray-500 ring-2 ring-white"
                      title="<mock-value-for-field-\\"login\\">"
                    >
                      <span
                        class="text-xs font-medium leading-none text-white cursor-default"
                      >
                        &lt;
                      </span>
                    </span>
                  </div>
                  <div
                    class="mt-2"
                  >
                    <p
                      class="text-sm text-gray-500"
                    >
                      &lt;mock-value-for-field-"repository"&gt; #&lt;mock-value-for-field-"number"&gt;
                    </p>
                  </div>
                </div>
              </div>
              <div>
                <a
                  href="/repos/<mock-value-for-field-\\"repository\\">/pull/<mock-value-for-field-\\"number\\">"
                  title="Open on Cody"
                >
                  <svg
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="9 18 15 12 9 6"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="py-6"
    >
      <button
        class="flex items-center block mx-auto bg-white hover:bg-gray-50 border-gray-300 text-gray-700 px-4 py-2 border shadow-sm text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500"
      >
        Load More
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReviewDashboardResults snapshot test 2`] = `
<DocumentFragment>
  <div
    class="max-w-7xl mx-auto sm:px-6 lg:px-8 py-6"
  >
    <div
      class="bg-white shadow overflow-hidden sm:rounded-md"
    >
      <ul
        class="divide-y divide-gray-200"
      >
        <li>
          <div
            class="flex items-center px-4 py-4 sm:px-6"
          >
            <div
              class="min-w-0 flex-1 flex items-center"
            >
              <div
                class="flex-shrink-0"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="18"
                    cy="18"
                    r="3"
                  />
                  <circle
                    cx="6"
                    cy="6"
                    r="3"
                  />
                  <path
                    d="M13 6h3a2 2 0 0 1 2 2v7"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="9"
                    y2="21"
                  />
                </svg>
              </div>
              <div
                class="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4"
              >
                <div>
                  <p
                    class="text-sm font-medium truncate"
                  >
                    <a
                      href="https://example.local"
                      rel="external"
                      target="_blank"
                      title="My Remote Title"
                    >
                      My Remote Title
                    </a>
                  </p>
                  <div
                    class="mt-2 sm:flex sm:justify-between"
                  >
                    <div
                      class="sm:flex"
                    >
                      <p
                        class="flex items-center text-sm text-gray-500"
                      >
                        <span
                          class="text-sm"
                          title="Why was I assigned?"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <circle
                              cx="12"
                              cy="12"
                              r="10"
                            />
                            <path
                              d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                            />
                            <line
                              x1="12"
                              x2="12.01"
                              y1="17"
                              y2="17"
                            />
                          </svg>
                        </span>
                        &lt;mock-value-for-field-"name"&gt;
                      </p>
                      <p
                        class="mt-2 flex items-center text-sm text-gray-500 sm:mt-0 sm:ml-6"
                      >
                        <span
                          class="text-sm"
                          title="Created at"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="18"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="4"
                            />
                            <line
                              x1="16"
                              x2="16"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="8"
                              x2="8"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="3"
                              x2="21"
                              y1="10"
                              y2="10"
                            />
                          </svg>
                        </span>
                        Nov 30, 2020
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden md:block"
                >
                  <div
                    class="-space-x-1"
                  >
                    <span
                      class="relative inline-flex items-center justify-center h-6 w-6 rounded-full bg-gray-500 ring-2 ring-white"
                      title="<mock-value-for-field-\\"login\\">"
                    >
                      <span
                        class="text-xs font-medium leading-none text-white cursor-default"
                      >
                        &lt;
                      </span>
                    </span>
                  </div>
                  <div
                    class="mt-2"
                  >
                    <p
                      class="text-sm text-gray-500"
                    >
                      &lt;mock-value-for-field-"repository"&gt; #&lt;mock-value-for-field-"number"&gt;
                    </p>
                  </div>
                </div>
              </div>
              <div>
                <a
                  href="/repos/<mock-value-for-field-\\"repository\\">/pull/<mock-value-for-field-\\"number\\">"
                  title="Open on Cody"
                >
                  <svg
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="9 18 15 12 9 6"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="py-6"
    >
      <button
        class="flex items-center block mx-auto bg-white hover:bg-gray-50 border-gray-300 text-gray-700 px-4 py-2 border shadow-sm text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500"
      >
        Load More
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReviewDashboardResults snapshot test 3`] = `
<DocumentFragment>
  <div
    class="max-w-7xl mx-auto sm:px-6 lg:px-8 py-6"
  >
    <div
      class="bg-white shadow overflow-hidden sm:rounded-md"
    >
      <ul
        class="divide-y divide-gray-200"
      >
        <li>
          <div
            class="flex items-center px-4 py-4 sm:px-6"
          >
            <div
              class="min-w-0 flex-1 flex items-center"
            >
              <div
                class="flex-shrink-0"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="18"
                    cy="18"
                    r="3"
                  />
                  <circle
                    cx="6"
                    cy="6"
                    r="3"
                  />
                  <path
                    d="M13 6h3a2 2 0 0 1 2 2v7"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="9"
                    y2="21"
                  />
                </svg>
              </div>
              <div
                class="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4"
              >
                <div>
                  <p
                    class="text-sm font-medium truncate"
                  >
                    <a
                      href="https://example.local"
                      rel="external"
                      target="_blank"
                      title="My Remote Title"
                    >
                      My Remote Title
                    </a>
                  </p>
                  <div
                    class="mt-2 sm:flex sm:justify-between"
                  >
                    <div
                      class="sm:flex"
                    >
                      <p
                        class="flex items-center text-sm text-gray-500"
                      >
                        <span
                          class="text-sm"
                          title="Why was I assigned?"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <circle
                              cx="12"
                              cy="12"
                              r="10"
                            />
                            <path
                              d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                            />
                            <line
                              x1="12"
                              x2="12.01"
                              y1="17"
                              y2="17"
                            />
                          </svg>
                        </span>
                        &lt;mock-value-for-field-"name"&gt;
                      </p>
                      <p
                        class="mt-2 flex items-center text-sm text-gray-500 sm:mt-0 sm:ml-6"
                      >
                        <span
                          class="text-sm"
                          title="Created at"
                        >
                          <svg
                            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="18"
                              rx="2"
                              ry="2"
                              width="18"
                              x="3"
                              y="4"
                            />
                            <line
                              x1="16"
                              x2="16"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="8"
                              x2="8"
                              y1="2"
                              y2="6"
                            />
                            <line
                              x1="3"
                              x2="21"
                              y1="10"
                              y2="10"
                            />
                          </svg>
                        </span>
                        Nov 30, 2020
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden md:block"
                >
                  <div
                    class="-space-x-1"
                  >
                    <span
                      class="relative inline-flex items-center justify-center h-6 w-6 rounded-full bg-gray-500 ring-2 ring-white"
                      title="<mock-value-for-field-\\"login\\">"
                    >
                      <span
                        class="text-xs font-medium leading-none text-white cursor-default"
                      >
                        &lt;
                      </span>
                    </span>
                  </div>
                  <div
                    class="mt-2"
                  >
                    <p
                      class="text-sm text-gray-500"
                    >
                      &lt;mock-value-for-field-"repository"&gt; #&lt;mock-value-for-field-"number"&gt;
                    </p>
                  </div>
                </div>
              </div>
              <div>
                <a
                  href="/repos/<mock-value-for-field-\\"repository\\">/pull/<mock-value-for-field-\\"number\\">"
                  title="Open on Cody"
                >
                  <svg
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="9 18 15 12 9 6"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="py-6"
    >
      <button
        class="flex items-center block mx-auto bg-white hover:bg-gray-50 border-gray-300 text-gray-700 px-4 py-2 border shadow-sm text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500"
      >
        <svg
          class="animate-spin -ml-1 mr-3 h-5 w-5 text-gray-700"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="opacity-25"
            cx="12"
            cy="12"
            r="10"
            stroke="currentColor"
            stroke-width="4"
          />
          <path
            class="opacity-75"
            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
            fill="currentColor"
          />
        </svg>
        Loading...
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReviewDashboardResults snapshot test with no results 1`] = `
<DocumentFragment>
  <div
    class="max-w-7xl mx-auto sm:px-6 lg:px-8 py-6"
  >
    <div
      class="bg-white shadow overflow-hidden sm:rounded-md"
    >
      <ul
        class="divide-y divide-gray-200"
      >
        <li>
          <div
            class="mx-auto text-sm font-medium px-4 py-4 sm:px-6"
          >
            <p
              class="flex items-center text-sm text-gray-500"
            >
              <svg
                class="flex-shrink-0 mr-1.5 h-5 w-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="20 12 20 22 4 22 4 12"
                />
                <rect
                  height="5"
                  width="20"
                  x="2"
                  y="7"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="22"
                  y2="7"
                />
                <path
                  d="M12 7H7.5a2.5 2.5 0 0 1 0-5C11 2 12 7 12 7z"
                />
                <path
                  d="M12 7h4.5a2.5 2.5 0 0 0 0-5C13 2 12 7 12 7z"
                />
              </svg>
              No pending reviews! Great job!
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
